contracts/engagement/src/contract.rs

```diff
@@ -91,18 +91,32 @@ impl EngagementContract {
             Ok(esc) => esc,
             Err(err) => return Err(err),
         };
+
+        if escrow.dispute_flag == false {
+            return Err("Escrow has been opened for dispute resolution");
+        }
     
-        let half_price_in_micro_usdc = (escrow.amount as i128) / 2;
         let usdc_client = TokenClient::new(&e, &usdc_contract);
-    
+
         let signer_balance = usdc_client.balance(&signer);
-        if signer_balance < half_price_in_micro_usdc {
+        
+        if escrow.balance > escrow.amount {
+            return Err("Escrow fully funded");
+        }
+
+        if amount_to_deposit > escrow.amount {
+            return Err("Amount to deposit is greather thant the escrow amount");
+        }
+
+        if signer_balance < amount_to_deposit {
             return Err(ContractError::SignerInsufficientFunds);
         }
+
+        let contract_address = e.current_contract();
```

Comment:
In this line, we are getting an error because of the method you are using to get the contract address. You should use `e.current_contract_address()` to correctly retrieve the contract address.